core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -1005,9 +1005,7 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties {
   test("Set Spark CallerContext") {
     val context = "test"
     new CallerContext(context).setCurrentContext()
-    if (CallerContext.callerContextEnabled) {
-      assert(s"SPARK_$context" === HadoopCallerContext.getCurrent.toString)
-    }
+    assert(s"SPARK_$context" === HadoopCallerContext.getCurrent.toString)
```

Comment:
Maybe consider adding an assert-not-null like in the #49858 test? If this isn't working, or if someone disables it in pom.xml, then `toString` will cause NPE instead of an assertion failure message.

Reply:
NPE is already enough to detect the failure. :)